ext/JarMain.java

```diff
@@ -123,14 +123,25 @@ private void listf(String directoryName, ArrayList<File> files) {
       // get all the files from a directory
       File[] fList = directory.listFiles();
       for (File file : fList) {
-        if (file.isFile()) {
+        String fileExtension = getFileExtension(file);
+        fileExtension = fileExtension.toLowerCase();
+        if (file.isFile() && fileExtension.equals("jar")) {
           files.add(file);
         } else if (file.isDirectory()) {
           listf(file.getAbsolutePath(), files);
         }
       }
     }
 
+    private String getFileExtension(File file) {
+      String name = file.getName();
+      try {
+          return name.substring(name.lastIndexOf(".") + 1);
```

Comment:
I will switch to the apache FilenameUtils impl for this:
http://commons.apache.org/proper/commons-io/javadocs/api-2.5/src-html/org/apache/commons/io/FilenameUtils.html

Going to include manually in class as private method(s) so as to avoid library dependency.